docs/DesignDocs/PluginArchitecture.md

```diff
@@ -0,0 +1,65 @@
+Plugin Architecture
+=====
+
+This document describes the principles behind the openenclave plugin
+architecture and establishes the framework for Third-party Testing of those
```

Comment:
Nit: "Third-party Testing" probably doesn't have any valence here as a proper noun, we should probably lower case.